chat_server/server_utils/socketio_utils.py

```diff
@@ -0,0 +1,71 @@
+# NEON AI (TM) SOFTWARE, Software Development Kit & Application Framework
+# All trademark and other rights reserved by their respective owners
+# Copyright 2008-2022 Neongecko.com Inc.
+# Contributors: Daniel McKnight, Guy Daniels, Elon Gasper, Richard Leeds,
+# Regina Bloomstine, Casimiro Ferreira, Andrii Pernatii, Kirill Hrymailo
+# BSD-3 License
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import json
+from time import time
+
+from typing import Optional, List
+from asyncio import Lock
+
+from chat_server.server_utils.api_dependencies import (CurrentUserModel,
+                                                       ListPersonasQueryModel)
+from chat_server.sio.server import sio
+
+
+_LOCK = Lock()
+
+
+async def notify_personas_changed(supported_llms: Optional[List[str]] = None):
+    """
+    Emit an SIO event for each LLM affected by a persona change. This sends a
+    complete set of personas rather than only the changed one to prevent sync
+    conflicts and simplify client-side logic.
+    :param supported_llms: List of LLM names affected by a transaction. If None,
+        then updates all LLMs listed in database configuration
+    """
+    from chat_server.blueprints.personas import list_personas
+    async with _LOCK:
+        resp = await list_personas(CurrentUserModel(_id="", nickname="",
```

Comment:
consider moving `listing personas` to the utilities so we bypass the authorization step

Reply:
I will extract logic from `list_personas` method into a utility and call directly here and in the `list_personas` method